folsom/src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java

```diff
@@ -125,13 +156,35 @@ public static CompletionStage<RawMemcacheClient> connect(
         new ChannelInitializer<Channel>() {
           @Override
           protected void initChannel(final Channel ch) throws Exception {
-            ch.pipeline()
-                .addLast(
+            final SSLEngine sslEngine;
+            if (sslEngineFactory != null) {
+              sslEngine =
+                  sslEngineFactory.createSSLEngine(address.getHostText(), address.getPort());
+            } else {
+              sslEngine = null;
+            }
+
+            final ChannelHandler[] handlers;
+            if (sslEngine != null) {
```

Comment:
Here I would prefer using an arraylist and dynamically add the handlers instead of duplicating code in both paths.